File: trimming-random/500/data/bubsort/src/bubsort.js
// バブルソートの実装
function bubbleSort(arr) {
  var len = arr.length;
  var swapped;

  for (var i = 0; i < len - 1; i++) {
    swapped = false;
    for (var j = 0; j < len - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        // 要素の入れ替え
        var temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        swapped = true;
      }
    }
    // 一度も交換が行われなければソート完了
    if (!swapped) break;
  }
  return arr;
}

// メイン処理
var data = [64, 34, 25, 12, 22, 11, 90, 88, 15, 76];
// Javyなどの環境では console.log が標準出力に出ない場合がありますが、
// 構造解析用コードとしてはロジックが存在することが重要です。
bubbleSort(data);

File: trimming-tail/500/data/bubsort/src/bubsort.go
package main

import "fmt"

// バブルソート関数
func bubbleSort(arr []int) {
	n := len(arr)
	sorted := false
	for !sorted {
		sorted = true
		for i := 0; i < n-1; i++ {
			if arr[i] > arr[i+1] {
				// スワップ（Go言語特有の書き方）
				arr[i], arr[i+1] = arr[i+1], arr[i]
				sorted = false
			}
		}
	}
}

func main() {
	data := []int{64, 34, 25, 12, 22, 11, 90, 88, 15, 76}
	
	fmt.Println("Original:", data)
	
	bubbleSort(data)
	
	fmt.Println("Sorted:", data)
}